### PR TITLE
Use pyo3-stub-gen to support IDE type hints

### PR DESCRIPTION
**Pull request by cardigan1008**
For #304 .

This PR introduces the use of `pyo3-stub-gen` to automatically generate `.pyi` files. These stub files provide type hints for Python IDEs.

Currently, the following modules are supported:

- [ ] Epoch
- [x] TimeScale
- [x] TimeSeries
- [ ] Duration
- [x] LatestLeapSeconds
- [x] LeapSecondsFile
- [x] Ut1Provider
- [x] PyHifitimeError
- [x] PyDurationError
- [x] PyParsingError

### CODE REVIEW

File: hifitime.pyi [hunk 1]
```diff
@@ -0,0 +1,81 @@
+# This file is automatically generated by pyo3_stub_gen
```
Comment (ChristopherRabotin) [inline hifitime.pyi:1]: It seems that the stub generator could not find any of the functions in any of the classes, so as-is this new `pyi` file won't help IDE users in autocompleting.

Comment (cardigan1008) [reply hifitime.pyi:1]: Yes. So maybe we can turn to another way to do this, like dora's?

Comment (ChristopherRabotin) [reply hifitime.pyi:1]: I like that dora's approach uses the package introspection, but I could not seem to get it to work correctly when I tried a few weeks (months?) ago. It kept on complaining that one of the return types was not defined although I thought I had. Feel free to give it a try of course!

**Comment by ChristopherRabotin** [comment]
I've implemented Dora's approach in #335 , so I'll close this PR.

**Pull request closed without merging**